### PR TITLE
Task 292 Feature Request: Allow configuration of initContainer resource requests and limits

### PR DESCRIPTION
**Pull request by dualBreath**
https://github.com/networkservicemesh/cmd-admission-webhook-k8s/issues/292
after a successful merge, update the version of the `admission-webhook-k8s` image tag using the latest tag from here: https://github.com/networkservicemesh/cmd-admission-webhook-k8s/pkgs/container/ci%2Fcmd-admission-webhook-k8s
PR to replace image tag: https://github.com/networkservicemesh/deployments-k8s/pull/9727

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -38,23 +38,27 @@ import (
...
+	Envs                    []string          `desc:"Additional Envs that should be appended for each Config.ContainerImages and Config.InitContainerImages" split_words:"true"`
+	CertFilePath            string            `desc:"Path to certificate" split_words:"true"`
+	KeyFilePath             string            `desc:"Path to RSA/Ed25519 related to Config.CertFilePath" split_words:"true"`
+	CABundleFilePath        string            `desc:"Path to cabundle file related to Config.CertFilePath" split_words:"true"`
+	OpenTelemetryEndpoint   string            `default:"otel-collector.observability.svc.cluster.local:4317" desc:"OpenTelemetry Collector Endpoint"`
+	ResourcesLimitsMemory   string            `default:"40Mi" desc:"Init container memory limits" split_words:"true"`
+	ResourcesLimitsCPU      string            `default:"100m" desc:"Init container CPU limits" split_words:"true"`
+	ResourcesRequestsMemory string            `default:"40Mi" desc:"Init container requests memory limits" split_words:"true"`
+	ResourcesRequestsCPU    string            `default:"100m" desc:"Init container requests CPU limits" split_words:"true"`
```
Comment (denis-tingaikin) [inline internal/config/config.go:57]: We need to change description and note that limits applying only and only for injected nsm containers.

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

File: main.go [hunk 2]
```diff
@@ -359,6 +360,19 @@ func (s *admissionWebhookServer) addResources(c *corev1.Container, r map[string]
 	}
 }
 
+func (s *admissionWebhookServer) addResourcesLimits(c *corev1.Container) {
```
[the remark this answers is not in the document]
Comment (dualBreath) [reply main.go]: Sounds reasonable, done

File: internal/config/config.go (hunk 1 above)
Comment (dualBreath) [reply internal/config/config.go]: Thanks, fixed

File: main.go [hunk 3]
```diff
@@ -359,6 +361,27 @@ func (s *admissionWebhookServer) addResources(c *corev1.Container, r map[string]
 	}
 }
 
+func (s *admissionWebhookServer) addNscLimits(c *corev1.Container) {
+	s.createResourcesLimits(c, s.config.NscLimitsCPU, s.config.NscRequestsCPU, s.config.NscLimitsMemory, s.config.NscRequestsMemory)
+}
+
+func (s *admissionWebhookServer) addNscInitLimits(c *corev1.Container) {
+	s.createResourcesLimits(c, s.config.NscInitLimitsCPU, s.config.NscInitRequestsCPU, s.config.NscInitLimitsMemory, s.config.NscInitRequestsMemory)
+}
```
Comment (denis-tingaikin) [inline main.go:370]: Seems like we can do this inline

```suggestion
```

File: main.go [hunk 4]
```diff
@@ -359,6 +361,27 @@ func (s *admissionWebhookServer) addResources(c *corev1.Container, r map[string]
...
+
+func (s *admissionWebhookServer) createResourcesLimits(c *corev1.Container, limitsCPU, requestsCPU, limitsMemory, requestsMemory string) {
```
Comment (denis-tingaikin) [inline main.go:372]:
```suggestion
func (s *admissionWebhookServer) addResourceLimits(c *corev1.Container) {
```

File: main.go [hunk 5]
```diff
@@ -359,6 +361,27 @@ func (s *admissionWebhookServer) addResources(c *corev1.Container, r map[string]
...
+	c.Resources = corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			"cpu":    resource.MustParse(limitsCPU),
```
Comment (denis-tingaikin) [inline main.go:375]:
```suggestion
			"cpu":    resource.MustParse(s.config.LimitsCPU),
```

File: internal/config/config.go [hunk 6]
```diff
@@ -51,6 +51,14 @@ type Config struct {
 	KeyFilePath           string            `desc:"Path to RSA/Ed25519 related to Config.CertFilePath" split_words:"true"`
 	CABundleFilePath      string            `desc:"Path to cabundle file related to Config.CertFilePath" split_words:"true"`
 	OpenTelemetryEndpoint string            `default:"otel-collector.observability.svc.cluster.local:4317" desc:"OpenTelemetry Collector Endpoint"`
+	NscInitLimitsMemory   string            `default:"80Mi" desc:"Injected nsm init container memory limits" split_words:"true"`
```
Comment (denis-tingaikin) [inline internal/config/config.go:54]:
```suggestion
	SidecarLimitsMemory   string            `default:"80Mi" desc: "Means lowerbound NSM sidecar memory limit (in k8s resource management units)" split_words:"true"`
```

File: main.go (hunk 3 above)
Comment (dualBreath) [reply main.go]: Fixed

File: main.go (hunk 4 above)
Comment (dualBreath) [reply main.go]: Fixed

File: internal/config/config.go (hunk 6 above)
Comment (dualBreath) [reply internal/config/config.go]: Fixed

File: main.go (hunk 5 above)
Comment (dualBreath) [reply main.go:367]: What is the idea behind this change?

Comment (dualBreath) [reply main.go]: As discussed personally, removed separate envs for the init container

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**